torchgen/utils.py

```diff
@@ -396,3 +396,52 @@ def _format(
     indent_str = " " * indent
     body = f", {delimiter}{curr_indent_str}".join(fields_str)
     return f"{start}{indent_str}{body}{end}"
+
+
+class NamespaceHelper:
+    """A helper for constructing the namespace open and close strings for a nested set of namespaces.
+
+    e.g. for namespace_str torch::lazy,
+
+    prologue:
+    namespace torch {
+    namespace lazy {
+
+    epilogue:
+    } // namespace lazy
+    } // namespace torch
+    """
+
+    def __init__(self, namespace_str: str):
+        # cpp_namespace can be a colon joined string such as torch::lazy
+        cpp_namespaces = namespace_str.split("::")
+        self.prologue_ = "\n".join([f"namespace {n} {{" for n in cpp_namespaces])
+        self.epilogue_ = "\n".join(
+            [f"}} // namespace {n}" for n in reversed(cpp_namespaces)]
+        )
+        if len(cpp_namespaces) == 1:
+            self.namespaces = []
+            self.entity_name_ = cpp_namespaces[0]
+        else:
+            self.namespaces = cpp_namespaces[:-1]
+            self.entity_name_ = cpp_namespaces[-1]
+
+    @property
+    def prologue(self) -> str:
+        return self.prologue_
+
+    @property
+    def epilogue(self) -> str:
+        return self.epilogue_
+
+    @property
+    def entity_name(self) -> str:
+        return self.entity_name_
+
+    # Only allow certain level of namespaces
+    def get_namespaces(self, max_level: int = 1, default: str = "") -> str:
```

Comment:
Please add a docstring describing how/when the default is used. Also document the format of the return string (i.e., a `::`-separated C++ namespace string, without a leading `::`)

In fact since this does return a single C++ namespace string, consider calling it the singular `get_namespace` or even `get_cpp_namespace` to make it more clear that it's C++ specific and doesn't return a list.